lib/trino-filesystem-manager/src/main/java/io/trino/filesystem/manager/FileSystemConfig.java

```diff
@@ -19,6 +19,7 @@ public class FileSystemConfig
 {
     private boolean hadoopEnabled = true;
     private boolean nativeS3Enabled;
+    private String cacheType;
```

Comment:
I don't think there should be connector specific cache implementations.
The cache should allow binding an optional implementation for `boolean isCacheEnabledFor(Location location)`, this can be used by connectors to filter out whatever files they don't want to cache. By default it can be a no-op implementation.

Reply:
Maybe we could allow binding an optional implementation of `Optional<CacheKey> cacheKeyFor(TrinoInputFile file)`? Then we could avoid fetching `lastModified` for the checkpoint files in the Delta Lake connector.

Reply:
bump, we should follow the way `IdentityCacheKey` is working